selector/selector.go

```diff
@@ -0,0 +1,42 @@
+package selector
+
+import (
+	"errors"
+
+	"github.com/micro/go-micro/v2/router"
+)
+
+var (
+	// DefaultSelector is the default selector
+	DefaultSelector = NewSelector()
+
+	// ErrNoneAvailable is returned by select when no routes were provided to select from
+	ErrNoneAvailable = errors.New("none available")
+)
+
+// Selector selects a route from a pool
+type Selector interface {
+	// Init a selector with options
+	Init(...Option) error
+	// Options the selector is using
+	Options() Options
+	// Select a route from the pool using the strategy
+	Select(...*router.Route) (*router.Route, error)
```

Comment:
Out of interest why is this variadic rather than a slice?

Reply:
Apart from that, interface looks good

Reply:
No real reason. It might be worth changing this actually as we could add Select options in the future.

Reply:
Yeah maybe change to slice then. I *think* most uses of this will be with an existing slice rather than cobbling together multiple items so it won't involve too many extra slice creations